File: app/src/main/java/com/example/my_thrd/MainActivity.java
package com.example.my_thrd;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Get references to UI elements
        EditText editTextNumber = findViewById(R.id.editTextNumber);
        Button buttonCalculateSquare = findViewById(R.id.buttonCalculateSquare);
        TextView textViewSquareResult = findViewById(R.id.textViewSquareResult);

        // Set a click listener on the button
        buttonCalculateSquare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Get the input number from the EditText
                String inputText = editTextNumber.getText().toString();

                // Validate the input
                if (inputText.isEmpty()) {
                    Toast.makeText(MainActivity.this, "Please enter a number", Toast.LENGTH_SHORT).show();
                    return;
                }

                try {
                    // Parse the input to a number and calculate the square
                    double number = Double.parseDouble(inputText);
                    double square = number * number;

                    // Display the result
                    textViewSquareResult.setText("Square: " + square);
                } catch (NumberFormatException e) {
                    // Handle invalid input
                    Toast.makeText(MainActivity.this, "Invalid number", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/calculator_app/MainActivity.kt
package com.example.calculator_app

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    // Declare UI elements as non-nullable
    private lateinit var etFirstNumber: EditText
    private lateinit var etSecondNumber: EditText
    private lateinit var tvResultValue: TextView
    private lateinit var btnAdd: Button
    private lateinit var btnSubtract: Button
    private lateinit var btnMultiply: Button
    private lateinit var btnDivide: Button
    private lateinit var btnClear: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Initialize UI elements
        etFirstNumber = findViewById(R.id.etFirstNumber)
        etSecondNumber = findViewById(R.id.etSecondNumber)
        tvResultValue = findViewById(R.id.tvResultValue)
        btnAdd = findViewById(R.id.btnAdd)
        btnSubtract = findViewById(R.id.btnSubtract)
        btnMultiply = findViewById(R.id.btnMultiply)
        btnDivide = findViewById(R.id.btnDivide)
        btnClear = findViewById(R.id.btnClear)

        // Set button click listeners
        btnAdd.setOnClickListener { calculate("+") }
        btnSubtract.setOnClickListener { calculate("-") }
        btnMultiply.setOnClickListener { calculate("*") }
        btnDivide.setOnClickListener { calculate("/") }
        btnClear.setOnClickListener { clearInputs() }
    }

    // Calculation method
    private fun calculate(operator: String) {
        val num1 = etFirstNumber.text.toString()
        val num2 = etSecondNumber.text.toString()

        if (num1.isEmpty() || num2.isEmpty()) {
            Toast.makeText(this, "Please enter both numbers", Toast.LENGTH_SHORT).show()
            return
        }

        try {
            val firstNumber = num1.toDouble()
            val secondNumber = num2.toDouble()
            var result = 0.0

            when (operator) {
                "+" -> result = firstNumber + secondNumber
                "-" -> result = firstNumber - secondNumber
                "*" -> result = firstNumber * secondNumber
                "/" -> if (secondNumber != 0.0) {
                    result = firstNumber / secondNumber
                } else {
                    Toast.makeText(this, "Cannot divide by zero", Toast.LENGTH_SHORT).show()
                    return
                }
            }

            tvResultValue.text = result.toString()
        } catch (e: NumberFormatException) {
            Toast.makeText(this, "Invalid input", Toast.LENGTH_SHORT).show()
        }
    }

    // Clear input fields and result
    private fun clearInputs() {
        etFirstNumber.setText("")
        etSecondNumber.setText("")
        tvResultValue.text = "0.00"
    }
}
